pkg/webhook/cm/plugins/sloconfig/resource_qos_checker.go

```diff
@@ -47,7 +59,119 @@ func NewResourceQOSChecker(oldConfig, newConfig *corev1.ConfigMap, needUnmarshal
+			if err := CheckSubNetQos(fldPath.Child("ClusterStrategy").Child("LSClass"), cfg.ClusterStrategy.LSClass.NetworkQOS); err != nil {
+				return err
+			}
+		}
+
+		if cfg.ClusterStrategy.LSRClass != nil {
+			if err := CheckSubNetQos(fldPath.Child("ClusterStrategy").Child("LSRClass"), cfg.ClusterStrategy.LSRClass.NetworkQOS); err != nil {
+				return err
+			}
+		}
+	}
+
+	// check netqos config is valid for each node strategy
+	for idx, nodeStrategy := range cfg.NodeStrategies {
+		if nodeStrategy.BEClass != nil {
+			if err := CheckSubNetQos(fldPath.Child("nodeStrategy").Child(strconv.Itoa(idx)).Child("BEClass"), nodeStrategy.BEClass.NetworkQOS); err != nil {
+				return err
+			}
+		}
+		if nodeStrategy.LSClass != nil {
+			if err := CheckSubNetQos(fldPath.Child("nodeStrategy").Child(strconv.Itoa(idx)).Child("LSClass"), nodeStrategy.LSClass.NetworkQOS); err != nil {
+				return err
+			}
+		}
+
+		if nodeStrategy.LSRClass != nil {
+			if err := CheckSubNetQos(fldPath.Child("nodeStrategy").Child(strconv.Itoa(idx)).Child("LSRClass"), nodeStrategy.LSRClass.NetworkQOS); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func CheckSubNetQos(fldPath *field.Path, qos *v1alpha1.NetworkQOSCfg) error {
+	if qos == nil {
+		return nil
+	}
+
+	if errs := ValidatePercentageOrQuantity(qos.IngressRequest, fldPath.Child("IngressRequest")); len(errs) > 0 {
+		return storage.NewInvalidError(errs)
+	}
+	if errs := ValidatePercentageOrQuantity(qos.IngressLimit, fldPath.Child("IngressLimit")); len(errs) > 0 {
+		return storage.NewInvalidError(errs)
+	}
+	if errs := ValidatePercentageOrQuantity(qos.EgressRequest, fldPath.Child("EgressRequest")); len(errs) > 0 {
+		return storage.NewInvalidError(errs)
+	}
+	if errs := ValidatePercentageOrQuantity(qos.EgressLimit, fldPath.Child("EgressLimit")); len(errs) > 0 {
+		return storage.NewInvalidError(errs)
+	}
+
+	return nil
+}
+
+// ValidatePercentageOrQuantity tests if a given value is a valid percentage or
+// quantity (defines in: https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/quantity.go#L100-L111).
+func ValidatePercentageOrQuantity(intOrPercent *intstr.IntOrString, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if intOrPercent == nil {
+		return allErrs
+	}
+
+	switch intOrPercent.Type {
+	case intstr.String:
+		allErrs = append(allErrs, ValidateQuantityField(intOrPercent.StrVal, fldPath)...)
+	case intstr.Int:
+		allErrs = append(allErrs, ValidatePercentageField(intOrPercent.IntValue(), fldPath)...)
+	default:
+		allErrs = append(allErrs, field.Invalid(fldPath, intOrPercent, "must be an integer just in 0-100 or quantity (e.g '50M')"))
+	}
+
+	return allErrs
+}
+
+// ValidateQuantityField validates that given value is quantity format, like 50M.
+func ValidateQuantityField(quantityStr string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if _, err := resource.ParseQuantity(quantityStr); err != nil {
```

Comment:
Why not consider returning Parse error directly? In addition, is the InvalidQuantityMsg of 50M appropriate, or should it be 50Mi?

Reply:
this is just for type checking, maybe 50M and 50Mi are both acceptable?

Reply:
I understand that the error returned by ParseQuantity already contains these error messages. If I understand correctly, there is no need to customize this kind of error message.

Reply:
Because I want to get the path to the invalid fields contained in the structure. such as:
```bash
ResourceQOSCfg.ClusterStrategy.LSRClass.IngressRequest: Invalid value: "50a": must be quantity format, like 50M
```
there are some similar checks in k8s: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/apps/validation/validation.go#L460-L473

Reply:
Validation percentage makes sense, but `InvalidQuantityMsg` is not required.